x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_lifecycle/modal.tsx

```diff
@@ -0,0 +1,444 @@
+  updateLifecycle,
+  updateInProgress,
+  getIlmPolicies,
+  ilmLocator,
+  definition,
+}: ModalOptions) {
+  const existingLifecycle = definition.stream.ingest.lifecycle;
+  const [selectedPolicy, setSelectedPolicy] = useState(
+    isIlmLifecycle(existingLifecycle) ? existingLifecycle.ilm.policy : undefined
+  );
+  const [policies, setPolicies] = useState<Array<EuiSelectableOption<IlmOptionData>>>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
+
+  useEffect(() => {
+    const phasesDescription = (phases: Phases) => {
+      const desc: string[] = [];
+      if (phases.hot) {
+        const rollover = phases.hot.actions.rollover;
+        const rolloverWhen = [
+          rollover?.max_age && 'max age ' + rollover.max_age,
+          rollover?.max_docs && 'max docs ' + rollover.max_docs,
+          rollover?.max_primary_shard_docs &&
+            'primary shard docs ' + rollover.max_primary_shard_docs,
+          rollover?.max_primary_shard_size &&
+            'primary shard size ' + rollover.max_primary_shard_size,
+          rollover?.max_size && 'index size ' + rollover.max_size,
+        ]
+          .filter(Boolean)
+          .join(' or ');
+        desc.push(`Hot (${rolloverWhen ? 'rollover when ' + rolloverWhen : 'no rollover'})`);
+      }
+      if (phases.warm) {
+        desc.push(`Warm after ${phases.warm.min_age}`);
+      }
+      if (phases.cold) {
+        desc.push(`Cold after ${phases.cold.min_age}`);
+      }
+      if (phases.frozen) {
+        desc.push(`Frozen after ${phases.frozen.min_age}`);
+      }
+      if (phases.delete) {
+        desc.push(`Delete after ${phases.delete.min_age}`);
+      } else {
+        desc.push('Keep data indefinitely');
+      }
+
+      return desc.join(', ');
+    };
+
+    setIsLoading(true);
+    getIlmPolicies()
+      .then((ilmPolicies) => {
+        const policyOptions = ilmPolicies.map(
+          ({ name, policy }): EuiSelectableOption<IlmOptionData> => ({
+            label: `${name}`,
+            searchableLabel: name,
+            checked: selectedPolicy === name ? 'on' : undefined,
+            data: {
+              phases: phasesDescription(policy.phases),
+            },
+          })
+        );
+
+        setPolicies(policyOptions);
+      })
+      .catch((error) => {
+        setErrorMessage('body' in error ? error.body.message : error.message);
+      })
+      .finally(() => setIsLoading(false));
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <EuiModal onClose={closeModal}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          {i18n.translate('xpack.streams.streamDetailLifecycle.attachIlm', {
+            defaultMessage: 'Attach a lifecycle policy to this stream',
+          })}
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        Select a pre-defined policy or visit{' '}
```

Comment:
Some non-i18n-ified strings in here

Reply:
fixed in https://github.com/elastic/kibana/pull/208461/commits/ec9c9d9d20f2921aa3ca7a9bcd040dcbe5c886f7